hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSourceStorage.scala

```diff
@@ -311,8 +310,7 @@ class TestMORDataSourceStorage extends SparkClientFunctionalTestHarness {
       // since it happens before the compaction is executed.
       // The deltacommit from the fifth and sixth round should write record positions in the log
       // files since it happens after the completed compaction
-      validateRecordPositionsInLogFiles(
-        metaClient, shouldContainRecordPosition = !enableNBCC && i != 4)
+      validateRecordPositionsInLogFiles(metaClient, shouldContainRecordPosition = !enableNBCC)
```

Comment:
can we also ensure the log header does not contain positions. 
also, if possible can we validate that key based merging is used and not position based merge

Reply:
I've added stricter validation on the base file instant time of the record positions in this test.  Also, without the fix in the file group record buffer to compare the base file instant times in the log header vs the actual base file (`HoodiePositionBasedFileGroupRecordBuffer#extractRecordPositions`), this test would fail because of the wrong results.  So we should be good now.